File: FibonacciRange.c
#include <stdio.h>

void printFibonacciInRange(int L, int R){
    int a=0, b=1, next;
    int printedOne = 0;

    if(a>=L) printf("%d\n", a);

    if(b>=L && b<=R){
    printf("%d\n", b);
    printedOne = 1;
    }

    next = a+b;
    while(next<=R){
        if(next>=L){
            if(next==1 && printedOne){
                a=b;
                b=next;
                next=a+b;
                continue;
            }
            printf("%d\n", next);
        }
        a=b;
        b=next;
        next=a+b;
    }
}

int main(){
    int L,R;
    scanf("%d %d", &L, &R);
    printFibonacciInRange(L,R);
    return 0;
}

File: FibonacciRange.swift
import Foundation

func printFibonacciInRange(L: Int, R: Int){
    var a=0, b=1
    var fibSet = Set<Int>()

    while a<=R{
        if a>=L{
            fibSet.insert(a)
        }
        let next = a+b
        a=b
        b=next
    }

    let sortedFibNumbers = Array(fibSet).sorted()
    for num in sortedFibNumbers{
        print(num)
    }
}

if let input  = readLine(){
    let values = input.split(separator: " ").compactMap{Int($0)}
    if values.count == 2{
        let L = values[0], R = values[1]
        printFibonacciInRange(L: L, R: R)
    }
}